parser/charset/encoding_gbk.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package charset
+
+import (
+	"strings"
+	"unicode"
+
+	"golang.org/x/text/encoding/simplifiedchinese"
+)
+
+// EncodingGBKImpl is the instance of encodingGBK
+var EncodingGBKImpl = &encodingGBK{encodingBase{enc: simplifiedchinese.GBK}}
+
+func init() {
+	EncodingGBKImpl.self = EncodingGBKImpl
+}
+
+// encodingGBK is GBK encoding.
+type encodingGBK struct {
+	encodingBase
+}
+
+// Name implements Encoding interface.
+func (e *encodingGBK) Name() string {
+	return CharsetGBK
+}
+
+// Peek implements Encoding interface.
+func (e *encodingGBK) Peek(src []byte) []byte {
+	charLen := 2
+	if len(src) == 0 || src[0] < 0x80 {
```

Comment:
seems 0x80 in golang lib is also a single byte

Reply:
Then it will not be compatible with MySQL in some cases.

Reply:
Same question here. Maybe we should change it to 0x81.

Reply:
just keep it as original, this pr is too big.😢

Reply:
@Defined2014 The difference can be more huge than we expected:

If we change it to 0x81, `一二三` is decoded to `涓€浜屼笁` instead of `涓?簩涓?`. The latter is compatible with MySQL.

https://github.com/pingcap/tidb/blob/b4df7ca4dcc4a851a1a04d4348dee2b153618a7b/parser/charset/encoding_test.go#L45-L58